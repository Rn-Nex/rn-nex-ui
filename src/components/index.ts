export * from './Accordion';
export * from './ActivityIndicator';
export * from './Avatar';
export * from './Badge';
export * from './Box';
export * from './Button';
export * from './Card';
export * from './CheckBox';
export * from './Chip';
export * from './Dialog';
export * from './Divider';
export * from './DropDown';
export * from './Image';
export * from './List';
export * from './Pagination';
export * from './Portal';
export * from './Radio';
export * from './Snackbar';
export * from './Stepper';
export * from './Switch';
export * from './TextField';
export * from './Transitions';
export * from './Typography';
