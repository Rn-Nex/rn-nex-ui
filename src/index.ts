export * from './components';
export * from './libraries';
export * from './utils';
