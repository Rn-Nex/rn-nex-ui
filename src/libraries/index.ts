export * from './themes/v2';
export * from './style';
