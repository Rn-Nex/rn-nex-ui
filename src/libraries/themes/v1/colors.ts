import { ColorShades, Theme } from './theme';

const initialLightTheme = {
  mode: 'light',
  primary: {
    50: '#FFEBEE',
    100: '#FFCDD2',
    200: '#EF9A9A',
    300: '#E57373',
    400: '#EF5350',
    500: '#F44336',
    600: '#E53935',
    700: '#D32F2F',
    800: '#C62828',
    900: '#B71C1C',
  },
  secondary: {
    50: '#E1F5FE',
    100: '#B3E5FC',
    200: '#81D4FA',
    300: '#4FC3F7',
    400: '#29B6F6',
    500: '#03A9F4',
    600: '#039BE5',
    700: '#0288D1',
    800: '#0277BD',
    900: '#01579B',
  },
  accent: {
    50: '#FCE4EC',
    100: '#F8BBD0',
    200: '#F48FB1',
    300: '#F06292',
    400: '#EC407A',
    500: '#E91E63',
    600: '#D81B60',
    700: '#C2185B',
    800: '#AD1457',
    900: '#880E4F',
  },
  deepOrange: {
    50: '#FBE9E7',
    100: '#FFCCBC',
    200: '#FFAB91',
    300: '#FF8A65',
    400: '#FF7043',
    500: '#FF5722',
    600: '#F4511E',
    700: '#E64A19',
    800: '#D84315',
    900: '#BF360C',
  },
  brown: {
    50: '#EFEBE9',
    100: '#D7CCC8',
    200: '#BCAAA4',
    300: '#A1887F',
    400: '#8D6E63',
    500: '#795548',
    600: '#6D4C41',
    700: '#5D4037',
    800: '#4E342E',
    900: '#3E2723',
  },
  grey: {
    50: '#FAFAFA',
    100: '#F5F5F5',
    200: '#EEEEEE',
    300: '#E0E0E0',
    400: '#BDBDBD',
    500: '#9E9E9E',
    600: '#757575',
    700: '#616161',
    800: '#424242',
    900: '#212121',
  },
  green: {
    50: '#E8F5E9',
    100: '#C8E6C9',
    200: '#A5D6A7',
    300: '#81C784',
    400: '#66BB6A',
    500: '#4CAF50',
    600: '#43A047',
    700: '#388E3C',
    800: '#2E7D32',
    900: '#1B5E20',
  },
  lightGreen: {
    50: '#F1F8E9',
    100: '#DCEDC8',
    200: '#C5E1A5',
    300: '#AED581',
    400: '#9CCC65',
    500: '#8BC34A',
    600: '#7CB342',
    700: '#689F38',
    800: '#558B2F',
    900: '#33691E',
  },
  lime: {
    50: '#F9FBE7',
    100: '#F0F4C3',
    200: '#E6EE9C',
    300: '#DCE775',
    400: '#D4E157',
    500: '#CDDC39',
    600: '#C0CA33',
    700: '#AFB42B',
    800: '#9E9D24',
    900: '#827717',
  },
  yellow: {
    50: '#FFFDE7',
    100: '#FFF9C4',
    200: '#FFF59D',
    300: '#FFF176',
    400: '#FFEE58',
    500: '#FFEB3B',
    600: '#FDD835',
    700: '#FBC02D',
    800: '#F9A825',
    900: '#F57F17',
  },
  amber: {
    50: '#FFF8E1',
    100: '#FFECB3',
    200: '#FFE082',
    300: '#FFD54F',
    400: '#FFCA28',
    500: '#FFC107',
    600: '#FFB300',
    700: '#FFA000',
    800: '#FF8F00',
    900: '#FF6F00',
  },
  orange: {
    50: '#FFF3E0',
    100: '#FFE0B2',
    200: '#FFCC80',
    300: '#FFB74D',
    400: '#FFA726',
    500: '#FF9800',
    600: '#FB8C00',
    700: '#F57C00',
    800: '#EF6C00',
    900: '#E65100',
  },
  red: {
    50: '#FFEBEE',
    100: '#FFCDD2',
    200: '#EF9A9A',
    300: '#E57373',
    400: '#EF5350',
    500: '#F44336',
    600: '#E53935',
    700: '#D32F2F',
    800: '#C62828',
    900: '#B71C1C',
  },
  pink: {
    50: '#FCE4EC',
    100: '#F8BBD0',
    200: '#F48FB1',
    300: '#F06292',
    400: '#EC407A',
    500: '#E91E63',
    600: '#D81B60',
    700: '#C2185B',
    800: '#AD1457',
    900: '#880E4F',
  },
  blueGrey: {
    50: '#ECEFF1',
    100: '#CFD8DC',
    200: '#B0BEC5',
    300: '#90A4AE',
    400: '#78909C',
    500: '#607D8B',
    600: '#546E7A',
    700: '#455A64',
    800: '#37474F',
    900: '#263238',
  },
  teal: {
    50: '#E0F2F1',
    100: '#B2DFDB',
    200: '#80CBC4',
    300: '#4DB6AC',
    400: '#26A69A',
    500: '#009688',
    600: '#00897B',
    700: '#00796B',
    800: '#00695C',
    900: '#004D40',
  },
  cyan: {
    50: '#E0F7FA',
    100: '#B2EBF2',
    200: '#80DEEA',
    300: '#4DD0E1',
    400: '#26C6DA',
    500: '#00BCD4',
    600: '#00ACC1',
    700: '#0097A7',
    800: '#00838F',
    900: '#006064',
  },
  lightBlue: {
    50: '#E1F5FE',
    100: '#B3E5FC',
    200: '#81D4FA',
    300: '#4FC3F7',
    400: '#29B6F6',
    500: '#03A9F4',
    600: '#039BE5',
    700: '#0288D1',
    800: '#0277BD',
    900: '#01579B',
  },
  white: {
    50: '#ffffff',
    100: '#F5F5F5',
    200: '#EEEEEE',
    300: '#E0E0E0',
    400: '#BDBDBD',
    500: '#9E9E9E',
    600: '#757575',
    700: '#616161',
    800: '#424242',
    900: '#212121',
  },
};

const createDarkColorTheme = (key: keyof typeof initialLightTheme) => {
  const colors = initialLightTheme[key] as ColorShades;
  const darkColors: any = {};
  const shades = ['50', '100', '200', '300', '400', '500', '600', '700', '800', '900'];
  const reverseShades = [...shades].reverse();

  shades.forEach((shade, index) => {
    const shadeIndex = reverseShades[index] as unknown as keyof ColorShades;
    darkColors[shade] = colors[shadeIndex];
  });

  return darkColors;
};

const initialDarkTheme: Theme = {
  mode: 'dark',
  primary: createDarkColorTheme('primary'),
  secondary: createDarkColorTheme('secondary'),
  accent: createDarkColorTheme('accent'),
  deepOrange: createDarkColorTheme('deepOrange'),
  brown: createDarkColorTheme('brown'),
  grey: createDarkColorTheme('grey'),
  green: createDarkColorTheme('green'),
  lightGreen: createDarkColorTheme('lightGreen'),
  lime: createDarkColorTheme('lime'),
  yellow: createDarkColorTheme('yellow'),
  amber: createDarkColorTheme('amber'),
  orange: createDarkColorTheme('orange'),
  red: createDarkColorTheme('red'),
  pink: createDarkColorTheme('pink'),
  blueGrey: createDarkColorTheme('blueGrey'),
  teal: createDarkColorTheme('teal'),
  cyan: createDarkColorTheme('cyan'),
  lightBlue: createDarkColorTheme('lightBlue'),
  white: createDarkColorTheme('white'),
};

export { initialLightTheme, initialDarkTheme };
