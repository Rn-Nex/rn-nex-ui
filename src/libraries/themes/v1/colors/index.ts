export * from './accent';
export * from './amber';
export * from './blueGrey';
export * from './brown';
export * from './cyan';
export * from './deepOrange';
export * from './green';
export * from './grey';
export * from './lightBlue';
export * from './lightGreen';
export * from './lime';
export * from './orange';
export * from './pink';
export * from './primary';
export * from './red';
export * from './secondary';
export * from './teal';
export * from './white';
export * from './yellow';
export * from './colors';
