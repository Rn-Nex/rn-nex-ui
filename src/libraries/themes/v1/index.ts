export * from './V2ThemeContext';
export * from './colors';
