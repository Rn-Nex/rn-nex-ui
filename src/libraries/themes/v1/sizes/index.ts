export * from './font';
export * from './fontWeight';
export * from './latterSpacing';
export * from './lingHeight';
export * from './spacing';
