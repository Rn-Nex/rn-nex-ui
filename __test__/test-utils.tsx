import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
