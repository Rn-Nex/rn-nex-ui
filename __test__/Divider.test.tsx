import React from 'react';
import { render } from './test-utils';
import { Divider, Text } from '../src';
import { View } from 'react-native';
import { spacing } from '../src/libraries/themes/v1/sizes';

describe('Divider Component', () => {
  const dividerMockTestId = 'divide-test-id';
  const dividerStartLineTestId = 'divide-start-line-test-id';
  const dividerEndLineTestId = 'divide-end-line-test-id';

  const mockTestLabel = 'test-label';

  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<Divider />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward ref correctly', () => {
    render(<Divider ref={mockRef} />);

    expect(mockRef.current).not.toBeNull();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should render the children correctly', () => {
    const { toJSON, getByText } = render(
      <Divider ref={mockRef}>
        <Text>{mockTestLabel}</Text>
      </Divider>,
    );

    const text = getByText(mockTestLabel);

    expect(toJSON).toMatchSnapshot();
    expect(text).toBeTruthy();
  });

  it('should add the start spacing when passed the variant (startSpacing)', () => {
    const { getByTestId } = render(<Divider ref={mockRef} testID={dividerMockTestId} variant="startSpacing" />);

    const divider = getByTestId(dividerMockTestId);
    expect(divider.props.style).toEqual(expect.objectContaining({ paddingLeft: spacing.lg }));
  });

  it('should add the end spacing when passed the variant (endSpacing)', () => {
    const { getByTestId } = render(<Divider ref={mockRef} testID={dividerMockTestId} variant="endSpacing" />);

    const divider = getByTestId(dividerMockTestId);
    expect(divider.props.style).toEqual(expect.objectContaining({ paddingRight: spacing.lg }));
  });

  it('should add the start and end spacing when passed the variant (middle)', () => {
    const { getByTestId } = render(<Divider ref={mockRef} testID={dividerMockTestId} variant="middle" />);

    const divider = getByTestId(dividerMockTestId);
    expect(divider.props.style).toEqual(expect.objectContaining({ paddingHorizontal: spacing.lg }));
  });

  it('should remove the start and end spacing when passed the variant (fullWidth)', () => {
    const { getByTestId } = render(<Divider ref={mockRef} testID={dividerMockTestId} variant="fullWidth" />);

    const divider = getByTestId(dividerMockTestId);
    expect(divider.props.style).not.toHaveProperty('paddingHorizontal');
    expect(divider.props.style).not.toHaveProperty('paddingRight');
    expect(divider.props.style).not.toHaveProperty('paddingLeft');
  });

  it('should change the border of the divider component when passed the borderColor prop', () => {
    const { getByTestId } = render(
      <Divider startLineTestId={dividerStartLineTestId} endLineTestId={dividerEndLineTestId} borderColor="red" />,
    );
    const startLine = getByTestId(dividerStartLineTestId);
    const endLine = getByTestId(dividerStartLineTestId);
    expect(startLine.props.style).toEqual(expect.objectContaining({ borderColor: 'red' }));
    expect(endLine.props.style).toEqual(expect.objectContaining({ borderColor: 'red' }));
  });

  it('should change the variant spacing of the divider component when passed the variantSpacing prop', () => {
    const { getByTestId } = render(
      <Divider ref={mockRef} testID={dividerMockTestId} variant="startSpacing" variantSpacing={20} />,
    );

    const divider = getByTestId(dividerMockTestId);
    expect(divider.props.style).toEqual(expect.objectContaining({ paddingLeft: 20 }));
  });

  it('should apply the dynamic style into the start line element', () => {
    const { getByTestId } = render(
      <Divider startLineTestId={dividerStartLineTestId} startLineStyles={{ backgroundColor: 'red' }} />,
    );

    const startLine = getByTestId(dividerStartLineTestId);
    expect(startLine.props.style).toEqual(expect.objectContaining({ backgroundColor: 'red' }));
  });

  it('should apply the dynamic style into the end line element', () => {
    const { getByTestId } = render(<Divider endLineTestId={dividerEndLineTestId} endLineStyles={{ backgroundColor: 'red' }} />);

    const endLine = getByTestId(dividerEndLineTestId);
    expect(endLine.props.style).toEqual(expect.objectContaining({ backgroundColor: 'red' }));
  });
});
