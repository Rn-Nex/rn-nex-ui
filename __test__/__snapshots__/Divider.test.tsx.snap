// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should render correctly 1`] = `[Function]`;

exports[`Divider Component should render the children correctly 1`] = `[Function]`;
